### PR TITLE
Adding a scheduled workflow for stale PRs.

### PR DESCRIPTION
**Pull request by ppedrot**
This is the Coq-specific part of https://github.com/coq/bot/pull/153.

### CODE REVIEW

**Review by ejgallego — APPROVED** [review]

**Comment by ppedrot** [comment]
I am not sure you can call curl directly from the workflow, the doc is not super clear about the environment. This probably needs testing.

**Comment by Zimmi48** [comment]
> I am not sure you can call curl directly from the workflow, the doc is not super clear about the environment. This probably needs testing.

I would expect that you can. @JasonGross is experienced and might be able to confirm.

File: .github/workflows/stale.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: Stale PRs
+
+on:
+  schedule:
+    # Everyday at 2am
+    - cron: '0 2 * * *'
```
Comment (JasonGross) [inline .github/workflows/stale.yml:6]: Do you need to specify that this should only happen on the master branch?  (It seems like maybe not, but I'm not sure...)

**Comment by JasonGross** [comment]
> > I am not sure you can call curl directly from the workflow, the doc is not super clear about the environment. This probably needs testing.
> 
> I would expect that you can. @JasonGross is experienced and might be able to confirm.

According to https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu2004-README.md (from https://docs.github.com/en/actions/using-github-hosted-runners/about-github-hosted-runners#supported-software), `curl` comes preinstlled.  And I can confirm that calling `curl` works fine; it's how I have run-coqbot-minimizer communicate with the bot

File: .github/workflows/stale.yml (hunk 1 above)
Comment (ppedrot) [reply .github/workflows/stale.yml:6]: No idea. How can we know?

Comment (Zimmi48) [reply .github/workflows/stale.yml:6]: I think it is reasonable to expect that schedules will only run for the default branch.

Comment (SkySkimmer) [reply .github/workflows/stale.yml:6]: https://docs.github.com/en/actions/reference/events-that-trigger-workflows#schedule
>Scheduled workflows run on the latest commit on the default or base branch.

**Comment by Zimmi48** [comment]
Since merging this PR will trigger the bot every day, let's wait a bit before merging:

- that the feedback at https://github.com/coq/bot/pull/153#issuecomment-875458440 is taken into account;
- that we trigger the bot manually a few more times to see if we get more feedback.

**Comment by Zimmi48** [comment]
We've triggered the bot manually many times. We are now in the phase where coqbot starts closing stale PRs.

@ppedrot FYI, here were some comments from PR authors related to this feature:

- https://github.com/coq/coq/pull/13916#issuecomment-889327842
- https://github.com/coq/coq/pull/11747#issuecomment-891012724

Some additional remark is that since this cron schedule is made to trigger notifications, maybe it would make more sense to make it run every work day at 2pm, instead of every day at 2am?

**Comment by Zimmi48** [comment]
>Some additional remark is that since this cron schedule is made to trigger notifications, maybe it would make more sense to make it run every work day at 2pm, instead of every day at 2am?

@ppedrot WDYT?

**Comment by ppedrot** [comment]
I thought that precisely by sending these at night you'd see them in the morning, but I have no particular opinion further that that, so do as you see fit.

**Comment by Zimmi48** [comment]
OK, and what about not triggering the bot during week-ends?

**Comment by ppedrot** [comment]
Also fine to me.

File: .github/workflows/stale.yml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+name: Stale PRs
+
+on:
+  schedule:
+    # Everyday at 2am
```
Comment (Zimmi48) [inline .github/workflows/stale.yml:5]:
```suggestion
    # Every workday at 2am
```

**Comment by Zimmi48** [comment]
I'll merge now so this can start running tonight.

@coqbot merge now

**Comment by Zimmi48** [comment]
coqbot was triggered twice tonight. It turns out that we have neglected that the scheduled workflow would also run on any fork whose `master` branch contains this commit. And some people even keep their forks up-to-date with a "pull bot".

Tonight it ran at: https://github.com/MaxMood96/coq/actions/runs/1196471591

But next time, it could also run at: https://github.com/SkySkimmer/coq/actions, https://github.com/nittaya1990/coq/actions, etc.

We need to fix this workflow to check that it is on the main repo and not any fork, but we might also need to either change the coqbot endpoint or introduce a secret, so that repositories that stay up-to-date with the current master do not keep triggering coqbot...

**Comment by Zimmi48** [comment]
This change should be enough to fix the first part:

```diff
diff --git a/.github/workflows/stale.yml b/.github/workflows/stale.yml
index dc6b98818e..c1d8bfd96f 100644
--- a/.github/workflows/stale.yml
+++ b/.github/workflows/stale.yml
@@ -4,6 +4,8 @@ on:
   schedule:
     # Every workday at 2am
     - cron: '0 2 * * 1-5'
+    # Do not run on forks (we want this request to happen only once every night)
+    - if: github.repository_owner == 'coq'
 
 jobs:
   stale_prs:
```

I'm awaiting feedback regarding:
>we might also need to either change the coqbot endpoint or introduce a secret, so that repositories that stay up-to-date with the current master do not keep triggering coqbot...

**Comment by ppedrot** [comment]
Isn't it a bit annoying to have to introduce a secret? I don't know if we already handle authentication ourselves in coqbot, but if not we'd need to reimplement our own system and/or hardwire some nonce in the code which is not pretty.

**Comment by Zimmi48** [comment]
It wouldn't be too difficult: we already have a secret set in the coqbot-app to check that the webhook we receive from it are legitimate (typically are merge request originating from an actual GitHub comment). The secret was generated once and is stored both on GitHub and in an environment variable on Heroku. We could do the same with the schedule (store the secret in a secret variable available to GitHub Actions in this repo and in a new environment variable on Heroku). What would be more problematic with this setup is if some other repository also needed to use this feature.

**Pull request merged**